openpype/hosts/fusion/plugins/publish/render_local.py

```diff
@@ -23,42 +21,28 @@ def process(self, instance):
         # This plug-in runs only once and thus assumes all instances
         # currently will render the same frame range
         context = instance.context
-        key = "__hasRun{}".format(self.__class__.__name__)
```

Comment:
I think this had a reason?

Reply:
You've moved the logic into `render_once` but the logic before also skipped collection of filenames.

Reply:
Moved it back to the process function.

Reply:
NOTE: Maybe it was right. I think @jakubjezek001 or @BigRoy could tell.

Reply:
Looking at how the old code worked, moving it back would result in the same behavior as before. But if Jakub och Roy says something else I'm more than happy to revert the change!

Reply:
This seems to be ok with me - as far as I had tested it, it works and that is important! ;)

Reply:
Actually no - I think this is now still broken. This logic should've moved to `render_once()`. Because as far as I can tell from the code, if you had multiple savers then all instances that are not the first processed saver instance would NOT have their representations added and thus would not be published/integrated.

@EmberLightVFX can you confirm the logic works for multiple savers from a single comp, do they all get published correctly?

Also, reading it over again - it should even store whether the render failed or not - so that the subsequent instances being processed now whether the render was completed or not. Currently it'd only raise the runtime error on the first instance.

Reply:
When reading the code again, yeah something doesn't feel complete correct.
It feels like some kind of for-loop is missing to go over each and every saver instance to append to data["representations"].

Right now it works the way the old code worked but updated. Did that work with comps with multiple savers?
I'll do some multi-saver tests when I get some minutes over!

Reply:
This pyblish plug-in is an InstancePlugin, so it runs **per instance**. Which I think is good.
However, since currently Fusion integration is set to have all saver instances always render the fusion comp frame range it's much more optimized to render all at once instead of one by one (which will be massively slower if sharing parts of the comp).

As such, the intent was to:
- have something that ensures it only renders once, for all instances.
- have each instance check whether the render succeeded or not - if it did, have it add the representation into the instance.

The other solution would be to make it a Pyblish ContextPlugin, to then find the right instances, etc. - but I'd refrain from doing that personally because a next step would be making it so that - with the new publisher - we can also customize the frame ranges per saver we want to actually render. 💪 For which we'll end up needing InstancePlugins I'd say.

> Right now it works the way the old code worked but updated. Did that work with comps with multiple savers?

I've no idea what happened where but before OpenPype in our pipeline (on which this logic is based) this definitely worked just fine. But in OP I think it must have been broken for a while somehow. Or I'm massively misreading the code. :)

Reply:
> However, since currently Fusion integration is set to have all saver instances always render the fusion comp frame range it's much more optimized to render all at once instead of one by one (which will be massively slower if sharing parts of the comp).

Ooohhh! Now I see what you mean! Yeah, that makes total sense that the check would be in the render_once function!
Gonna make a new PR with the fix and add some comments about it in the file itself for future readers.